### PR TITLE
EVG-2354: refactor cli handling

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: artifacts-abcdef-rest_task_variant_task_one/robots.txt [hunk 1]
```diff
@@ -0,0 +1,276 @@
+User-agent: *
```
Comment (john-m-liu) [inline artifacts-abcdef-rest_task_variant_task_one/robots.txt]: What is this file for? There are a couple more like this

Comment (bsamek) [inline artifacts-abcdef-rest_task_variant_task_one/robots.txt]: Should this file be committed?

File: artifacts-abcdef-rest_task_variant_task_one/robots_(1).txt [hunk 2]
```diff
@@ -0,0 +1,276 @@
+User-agent: *
```
Comment (bsamek) [inline artifacts-abcdef-rest_task_variant_task_one/robots_(1).txt]: Don't commit

File: artifacts-abcdef-rest_task_variant_task_two/humans.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.
```
Comment (bsamek) [inline artifacts-abcdef-rest_task_variant_task_two/humans.txt]: Don't commit

File: .gitignore [hunk 4]
```diff
@@ -61,3 +61,4 @@ creds.yml
 fabfile.py
 *.pyc
 .idea
+evergreen
```
Comment (bsamek) [inline .gitignore]: Is this just for convenience, or are we actually building a binary in the root of the repo?

File: artifacts-abcdef-rest_task_variant_task_one/robots.txt (hunk 1 above)
Comment (tychoish) [reply artifacts-abcdef-rest_task_variant_task_one/robots.txt]: I believe they're detritus from the smoke test.

File: operations/list.go [hunk 5]
```diff
@@ -0,0 +1,301 @@
+package operations
```
Comment (john-m-liu) [inline operations/list.go]: compile errors:

operations/list.go:73:23: too many arguments in call to listAliases
	have (context.Context, string, string, string, string)
	want (context.Context, string, string, string)
operations/list.go:220:2: undefined: comm
operations/list.go:228:20: undefined: ac
operations/list.go:229:18: undefined: v2
operations/list.go:229:38: undefined: lc
operations/list.go:238:18: undefined: comm
operations/list.go:250:1: missing return at end of function

File: operations/patch.go [hunk 6]
```diff
@@ -0,0 +1,157 @@
+package operations
```
Comment (john-m-liu) [inline operations/patch.go]: compile errors: 
operations/patch.go:67:4: undefined: comm
operations/patch.go:76:59: undefined: comm
operations/patch.go:143:43: not enough arguments in call to params.validatePatchCommand
	have (*ClientSettings, *legacyClient)
	want (context.Context, *ClientSettings, *legacyClient, client.Communicator)
operations/patch.go:143:43: too many errors

File: .gitignore (hunk 4 above)
Comment (tychoish) [reply .gitignore]: Mostly the former. It's now reasonable to type `go build main/evergreen.go` and you end up with an evergreen binary in the repo (which is reasonable, and we shouldn't commit it.)

**Comment by bsamek** [comment]
I'm fine with this once tests pass and we verify it in staging. Should we manual test this a bit as well?

File: operations/cli_integration_test.go [hunk 7]
```diff
@@ -1,4 +1,4 @@
-package cli
+package operations
```
Comment (john-m-liu) [inline operations/cli_integration_test.go]: compile error:
operations/cli_integration_test.go:134:12: cannot use []string literal (type []string) as type string in field value
operations/cli_integration_test.go:135:4: cannot use false (type bool) as type []string in field value
operations/cli_integration_test.go:135:4: too few values in struct initializer
operations/cli_integration_test.go:346:14: cannot use []string literal (type []string) as type string in field value
operations/cli_integration_test.go:347:6: cannot use false (type bool) as type []string in field value
operations/cli_integration_test.go:347:6: too few values in struct initializer
operations/cli_integration_test.go:401:14: cannot use []string literal (type []string) as type string in field value
operations/cli_integration_test.go:402:6: cannot use false (type bool) as type []string in field value
operations/cli_integration_test.go:402:6: too few values in struct initializer
operations/cli_integration_test.go:414:14: cannot use []string literal (type []string) as type string in field value
operations/cli_integration_test.go:414:14: too many errors

File: operations/host_spawn.go [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+				return errors.Wrap(err, "problem contacting evergreen service")
+			}
+
+			grip.Infof("Spawn host created with ID '%s'. Visit the hosts page in Evergreen to check on its status.", host.Id)
+			return nil
+		},
+	}
+
+}
+func hostlist() cli.Command {
```
Comment (richardsamuels) [inline operations/host_spawn.go]: nit, should be hostList

**Pull request merged**